test/coverage/datamodel/PayloadBuilderTest.cpp

```diff
@@ -23,16 +26,667 @@ class TestFixture : public testing::Test {
+
+	// Act
+	Payload payload(std::move(serialized));
+
+	// Assert
+	auto [payloadData, payloadFormat] = payload.build();
+	EXPECT_EQ(payloadData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+}
+
+// Create payload of RValue Serialized with invalid format
+TEST_F(PayloadTest, ConstructorInvalidFormatForRValueSerializedPayloadTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    static_cast<uprotocol::v1::UPayloadFormat>(9999);
+	Payload::Serialized serialized =
+	    std::make_tuple(testStringPayload_, format);
+
+	// Act and Assert
+	EXPECT_THROW(Payload payload(std::move(serialized));, std::out_of_range);
+}
+
+// Create empty RValue Serialized payload and build.
+TEST_F(PayloadTest, EmptyRValueSerializedPayloadTest) {
+	// Arrange
+	std::string serializedData = "";
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+	Payload::Serialized serialized = std::make_tuple(serializedData, format);
+
+	// Act
+	Payload payload(std::move(serialized));
+
+	// Assert
+	EXPECT_THROW(payload.build(), Payload::PayloadMoved);
+}
+
+// Create RValue Serialized and move payload object test
+TEST_F(PayloadTest, RValueSerializedMovePayloadTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+	Payload::Serialized serialized =
+	    std::make_tuple(testStringPayload_, format);
+
+	// Act
+	Payload payload(std::move(serialized));
+	auto [payloadData, payloadFormat] = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(payloadData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+}
+
+// Create RValue Serialized payload and call move() twice on payload.
+TEST_F(PayloadTest, RValueSerializedMovePayloadTwiceTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+	Payload::Serialized serialized =
+	    std::make_tuple(testStringPayload_, format);
+
+	// Act
+	Payload payload(std::move(serialized));
+	auto [payloadData, payloadFormat] = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(payloadData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+
+	// Check that the original payload has been moved
+	EXPECT_THROW(std::move(payload).move(), Payload::PayloadMoved);
+}
+
+// Create RValue Serialized payload and call move() before calling build()
+TEST_F(PayloadTest, RValueSerializedMovePayloadBeforeBuildTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+	Payload::Serialized serialized =
+	    std::make_tuple(testStringPayload_, format);
+
+	// Act
+	Payload payload(std::move(serialized));
+	auto [payloadData, payloadFormat] = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(payloadData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+
+	// Check that the original payload has been moved
+	EXPECT_THROW(payload.build(), Payload::PayloadMoved);
+}
```

Comment:
It really isn't necessary to test `move()` and `build()` for each constructor. They are decoupled from the constructors by the `payload_` tuple.

Reply:
Ok I will just test this for only 1 constructor and will remove it for other constructors